File: LeetCode/cpp/2788_Split_Strings_by_Separator.cpp
class Solution {
public:
	vector<string> splitWordsBySeparator(vector<string>& words, char separator) {
		vector<string> answer;

		string s;
		for (int i = 0; i < words.size(); i++) {
			for (int j = 0; j < words[i].length(); j++) {
				if (words[i][j] == separator) {
					if (!s.empty()) {
						answer.push_back(s);
						s.clear();
					}
				}
				else {
					s += words[i][j];
				}
			}

			if (!s.empty()) {
				answer.push_back(s);
				s.clear();
			}
		}

		return answer;
	}
};

File: LeetCode/java/2788_Split_Strings_by_Separator.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<String> splitWordsBySeparator(List<String> words, char separator) {
        List<String> answer = new ArrayList<>();

        StringBuilder sb = new StringBuilder();
        for (String word : words) {
            for (int i = 0; i < word.length(); i++) {
                if (word.charAt(i) == separator) {
                    if (!sb.isEmpty()) {
                        answer.add(sb.toString());
                        sb.setLength(0);
                    }
                } else {
                    sb.append(word.charAt(i));
                }
            }

            if (!sb.isEmpty()) {
                answer.add(sb.toString());
                sb.setLength(0);
            }
        }

        return answer;
    }
}